### ISSUE TITLE
Launch Profile UI does not persist properties added by custom launch profile

### ISSUE THREAD

**Issue by alwu-msft** [dotnet/project-system#7384]
*TL;DR from @tmeschter*: Currently every property you wish to read or write in the Launch Profiles UI needs to be handled by some implementation of `ILaunchProfileExtensionValueProvider`. We should implement a fallback such that any property **not** handled by an `ILaunchProfileExtensionValueProvider` is simply passed through "as-is" and stored in the `ILaunchProfile.OtherSettings` dictionary. This would allow simple scenarios to have a very simple implementation.

*Original issue report continues below.*

**Visual Studio Version**: 17.0.0 Preview 2.0 [31430.505.main] (Dev17.IntPreview)

**Summary**: 
After following the [documentation](https://github.com/dotnet/project-system/blob/main/docs/repo/property-pages/how-to-add-a-new-launch-profile-kind.md) on adding a custom launch profile, I found that the Launch Profile UI fails to persist properties defined by the custom launch profile. Oddly, a newly-added property whose name matches that of one of the "standard" properties (`RemoteDebugEnabled`) *does* get persisted to `launchSettings.json`.

**Steps to Reproduce**:
[Repro project](https://github.com/dotnet/project-system/files/6751770/WpfApp1.zip)
New launch profile is defined in `<rootdir>\MyCustomLaunchProfile.xaml` and imported into the project via
```xml
<ItemGroup>
    <PropertyPageSchema Include="..\MyCustomLaunchProfile.xaml">
        <Context>Project</Context>
    </PropertyPageSchema>
</ItemGroup>
```
A `launchSettings.json` with a predefined profile using the newly defined launch command (`MyCustomLaunchProfile`) is included for convenience.

1. Open the Launch Profiles UI
2. (only needed if included `launchSettings.json` is deleted) Create a new profile based on "My Custom Launch Profile"
3. Change the value of each of the three listed properties (either by clicking on the checkbox or entering text into the TextBox).
4. Close and reopen the Launch Profiles UI

**Expected Behavior**:
Both checkboxes should be checked, and text entered into "Test String Property" should be preserved. Opening `launchSettings.json` will show values set on the three properties.

**Actual Behavior**:
The "Test bool property" checkbox will never be checked (if you click on it quickly enough you might notice a checkmark flicker in and out of existence, almost as if something is clobbering the checkbox's state). The text in the "Test string property" TextBox will be cleared as soon as the TextBox loses focus. Only the `RemoteDebugEnabled` property will have a value set in `launchSettings.json`.

**User Impact**:
This makes it impossible for a project system to add a functional custom launch profile.

**Comment by tmeschter** [comment]
@alwu-msft Currently every property you wish to read or write needs to be handled by some implementation of [`ILaunchProfileExtensionValueProvider`](https://github.com/dotnet/project-system/blob/main/src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Properties/LaunchProfiles/ILaunchProfileExtensionValueProvider.cs). There are various reasons for this:
1. A property may have an effective default value even when it is not present in the launch profile.
2. The value of a property _A_ may require arbitrary calculation rather than being read directly from the launch profile--e.g., it depends on the value of a different property _B_, or it needs to interact with some other component entirey.
3. Setting the value of a property may require the arbitrary calculation as well--e.g., if the user sets _A_ to true, you may want to automatically set _B_ to false.

So what you're seeing is that those new properties have no `ILaunchProfileExtensionValueProvider` and the values are just being dropped because we don't need to handle them.

I think the right thing to do is to simply persist the value "as is" to the `ILaunchProfile.OtherSettings` dictionary in the case where there is no `ILaunchProfileExtensionValueProvider` to handle the property. Would that work for you?

**Comment by alwu-msft** [comment]
Ah, the documentation was missing that part. :) Your explanation makes sense, as does your proposal to eliminate boilerplate for the simple case. Thanks!

**Comment by tmeschter** [comment]
Yeah, I'm behind on the documentation. :-(

**Comment by jjmew** [comment]
@tmeschter Please let us know if you need someone else to take on this

**Comment by tmeschter** [comment]
I have a partial solution for this one so I'd like to keep it on me for the moment.
